pkg/kapp/app/recorded_app.go

```diff
@@ -77,6 +77,51 @@ func (a *RecordedApp) UpdateUsedGVs(gvs []schema.GroupVersion) error {
 	})
 }
 
+func (a *RecordedApp) UsedGKs() ([]schema.GroupKind, error) {
+	meta, err := a.meta()
+	if err != nil {
+		return nil, err
+	}
+
+	return meta.UsedGKs, nil
+}
+
+func (a *RecordedApp) UpdateUsedGKs(gvks []schema.GroupKind, merge bool) ([]schema.GroupKind, error) {
+	gksByGK := map[schema.GroupKind]struct{}{}
+	var uniqGKs []schema.GroupKind
+
+	if merge {
+		usedGKs, err := a.UsedGKs()
```

Comment:
Ideally, we should not use one public method inside another public method.

Reply:
As of now we are using it basically as an internal function so I do not see any harm in making this internal.
The alternative would be to get the meta and then get the used GKs from that. I was trying to avoid duplication.

I am considering two possibilities now:
1. Update cache and use UsedGK to fetch GKs in `deploy.go`. We could have `usedGK()` and `UsedGK()` in this case (Rohit also noted this in his comments)
2. Make `UsedGKs` non exported and continue returning GKs with `UpdateUsedGKs()`

Reply:
I think the first approach makes more sense.

Reply:
Incorporated the first approach 🚀